### PR TITLE
Add methods to walk Endpoints and Networks

### PR DESCRIPTION
**Pull request by aboch**
Add methods to walk Endpoints and Networks

### CODE REVIEW

**Comment by mrjana** [comment]
LGTM

File: network.go [hunk 1]
```diff
@@ -50,9 +50,16 @@ import (
...
+
 	// Create a new network. The options parameter carries network specific options.
 	// Labels support will be added in the near future.
 	NewNetwork(d *NetworkDriver, name string, options interface{}) (Network, error)
+
+	// Networks returns the list of Network(s) managed by this controller.
+	Networks() []Network
+
+	// NetworksWalk uses the provided function to walk the Network(s) managed by this controller.
+	NetworksWalk(walker NwWalker)
```
Comment (mavenugo) [inline network.go]: Minor nit. Can you rename the method name to just `Walk` ?

File: network.go [hunk 2]
```diff
@@ -72,13 +79,20 @@ type Network interface {
 	// Labels support will be added in the near future.
 	CreateEndpoint(name string, sboxKey string, options interface{}) (Endpoint, error)
 
-	// Endpoints returns the list of Endpoint in this network.
+	// Endpoints returns the list of Endpoint(s) in this network.
 	Endpoints() []Endpoint
 
+	// EndpointsWalk uses the provided function to walk the Endpoints
+	EndpointsWalk(walker EpWalker)
```
Comment (mavenugo) [inline network.go]: Minor nit. Can you rename the method name to just `Walk` ?

File: network.go (hunk 1 above)
Comment (aboch) [inline network.go]: What about in future we want to walk the drivers ?

Comment (dave-tucker) [inline network.go]: Either way I think it would be `WalkNetworks`

Comment (aboch) [inline network.go]: That works, thanks. Will change it.

File: network.go [hunk 3]
```diff
@@ -74,13 +81,20 @@ type Network interface {
...
+	// WalkEndpoints uses the provided function to walk the Endpoints
+	WalkEndpoints(walker EpWalker)
+
 	// Delete the network.
 	Delete() error
 }
 
+// NwWalker is a client provided function which will be used to walk the Networks.
+// When the function returns true, the walk will stop
+type NwWalker func(nw Network) bool
```
Comment (mavenugo) [inline network.go]: Being an API exposed to the applications, shorter forms of names such as Nw and Ep as part of the Type doesn't feel right. This is just a preference. I will check with @dave-tucker to see what he thinks about this.

**Comment by mavenugo** [comment]
LGTM except for a minor nit on the naming of NwWalker and EpWalker. @dave-tucker can you pls comment on that ?

File: network.go (hunk 3 above)
Comment (aboch) [inline network.go]: That's true. I will try to come out with a better name.

Comment (dave-tucker) [inline network.go]: `SkyWalker` - just kidding :)
being verbose doesn't hurt here - e.g `EndpointWalker`, `NetworkWalker`

**Comment by mavenugo** [comment]
Thanks for addressing the comments. LGTM.

**Pull request merged**